.github/workflows/build-and-check.yml

```diff
@@ -9,61 +9,39 @@ on:
       - master
 
 jobs:
+  validation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: gradle/wrapper-validation-action@v1
+        with:
+          min-wrapper-count: 2
+
   build:
     strategy:
       matrix:
         fail-fast: false
-        os: [ubuntu-latest, windows-latest]
+        os: [ ubuntu-latest, windows-latest ]
     # Windows is failing with "out of memory" errors.
     continue-on-error: ${{ matrix.os == 'windows-latest' }}
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - name: set up JDK 1.8
-        uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
         with:
           java-version: 8
           distribution: 'zulu'
-      - name: Restore Gradle caches
-        uses: actions/cache@v2
-        with:
-          path: |
-            ~/.gradle/caches/modules-2
-            ~/.gradle/caches/files-2.1
-            ~/.gradle/caches/metadata-2.96
-          key: ${{ runner.os }}-gradle-cache-${{ hashFiles('**/*.gradle.kts') }}-${{ hashFiles('**/gradle.properties') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-cache-
-      # Inspired by https://github.com/actions/cache/issues/432#issuecomment-740376179
-      - name: Restore TestKit cache
-        uses: actions/cache@v2
-        with:
-          path: |
-            plugin/.gradle-test-kit/caches/modules-2
-            plugin/.gradle-test-kit/caches/files-2.1
-            plugin/.gradle-test-kit/caches/metadata-2.96
-          key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/gradlew') }}-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/gradlew') }}-
-            ${{ runner.os }}-gradle-wrapper-
-      - name: Restore Gradle wrapper
-        uses: actions/cache@v2
-        with:
-          path: |
-            ~/.gradle/wrapper
-          key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/gradlew') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-wrapper-
+      - uses: gradle/gradle-build-action@v2
```

Comment:
https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.

Reply:
I haven't had time to make this change myself, but I was intending to. I agree this is the right thing to do. Thanks for making the conversion here!